flytekit/common/tasks/presto_task.py

```diff
@@ -0,0 +1,144 @@
+from __future__ import absolute_import
+
+from google.protobuf.json_format import MessageToDict as _MessageToDict
+
+from flytekit.common import constants as _constants
+from flytekit.common.tasks import task as _base_task
+from flytekit.models import (
+    interface as _interface_model
+)
+from flytekit.models import literals as _literals, types as _types, \
+    task as _task_model
+
+from flytekit.common import interface as _interface
+import datetime as _datetime
+from flytekit.models import presto as _presto_models
+
+
+class SdkPrestoTask(_base_task.SdkTask):
+    """
+    This class includes the logic for building a task that executes as a Presto task.
+    """
+
+    def __init__(
+            self,
+            query,
+            output_schema,
+            routing_group=None,
+            catalog=None,
+            schema=None,
+            task_inputs=None,
+            interruptible=False,
+            discoverable=False,
+            discovery_version=None,
+            retries=1,
+            timeout=None,
+    ):
+        """
+        :param Text query: Presto query specification
+        :param flytekit.common.types.schema.Schema output_schema: Schema that represents that data queried from Presto
+        :param Text routing_group: The routing group that a Presto query should be sent to for the given environment
+        :param Text catalog: The catalog to set for the given Presto query
+        :param Text schema: The schema to set for the given Presto query
+        :param dict[Text,flytekit.common.types.base_sdk_types.FlyteSdkType] task_inputs: Optional inputs to the Presto task
+        :param bool discoverable:
+        :param Text discovery_version: String describing the version for task discovery purposes
+        :param int retries: Number of retries to attempt
+        :param datetime.timedelta timeout:
+        """
+
+        # Set as class fields which are used down below to configure implicit
+        # parameters
+        self._routing_group = routing_group or ""
+        self._catalog = catalog or ""
+        self._schema = schema or ""
```

Comment:
schema should always be non-null and can't default to a string.

Reply:
No no, this is a poor name that we unfortunately didn't realize earlier.  This is not the flytekit schema, this is a Presto schema.  @lu4nm3 if these three inputs are `None` will the backend plugin work correctly?

Reply:
Oops my bad.  @wild-endeavor is correct. This is a Presto "schema" which is equivalent to a  "db" in other database  systems

Reply:
Is it too late to rename?

Reply:
Not sure. Even if it wasn't, I would rather do the renaming as a separate PR if possible  as I feel this one has been dragged out for a while